pytest_mitmproxy.py

```diff
@@ -1 +1,76 @@
 # -*- coding: utf-8 -*-
+import re
+import os
+import shutil
+import pytest
+from contextlib import contextmanager
+from subprocess import Popen, CalledProcessError, TimeoutExpired
+from tempfile import TemporaryFile
+from urllib.parse import urlparse
+
+# clean the dumps each run
+if os.path.isdir("traffic_dumps"):
+    shutil.rmtree("traffic_dumps")
+
+
+@contextmanager
+def new_proxy(name, scope, flt=None):
+    ''' Start a new mitmdump process and
+    return the address for clients to connect
+
+    The captured content will be written to
+    ./<dump_dir>/<scope>/<name>
+    '''
+    dump_dir = "traffic_dumps"
```

Comment:
Make this configurable (cmd line flag)?
Also I'd use a name reflective of the underlying tool.
A pytest plugin user will probably better understand what the dir is for when they see `mitmproxy` in the dir name.